org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/json/JSonLanguageServer.java

```diff
@@ -25,29 +25,55 @@
 import java.util.Map;
 
 import org.eclipse.core.runtime.FileLocator;
-import org.eclipse.core.runtime.IConfigurationElement;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Platform;
 import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.content.IContentType;
+import org.eclipse.core.runtime.content.IContentTypeManager;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.jface.util.IPropertyChangeListener;
+import org.eclipse.jface.util.PropertyChangeEvent;
+import org.eclipse.lsp4e.LanguageServersRegistry;
+import org.eclipse.lsp4e.LanguageServersRegistry.LanguageServerDefinition;
+import org.eclipse.lsp4e.LanguageServiceAccessor;
 import org.eclipse.lsp4e.server.ProcessStreamConnectionProvider;
 import org.eclipse.lsp4j.InitializeResult;
 import org.eclipse.lsp4j.jsonrpc.messages.Message;
 import org.eclipse.lsp4j.jsonrpc.messages.ResponseMessage;
 import org.eclipse.lsp4j.services.LanguageServer;
 import org.eclipse.wildwebdeveloper.Activator;
+import org.eclipse.wildwebdeveloper.SchemaAssociationsPreferenceInitializer;
 import org.eclipse.wildwebdeveloper.embedder.node.NodeJSManager;
 
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+@SuppressWarnings("restriction")
 public class JSonLanguageServer extends ProcessStreamConnectionProvider {
 
-	private final static String SCHEMA_EXT = "org.eclipse.wildwebdeveloper.json.schema"; //$NON-NLS-1$
```

Comment:
Now this extension point is here, we can't remove it too quickly as it would break some adopters. The code that consumes it needs to remain, and be merged with the new code reading preferences. We can then consider deprecating the extension point later.

Reply:
Ah I see, that's a fair point!
As the extension point offers patterns and no contentTypes, is it okay to pass the associations from the extension point again directly to the `JSonLanguageServer`?
I think it might be a bit misleading to include them in the preference table with its different pattern / url combination.
I would then add additional info line on the preference page, to inform the user that the schema associations from the extension points are not included but still applying.
What do you think?

Reply:
you're right, sticking to JSonLanguageServer seems more correct and less confusing.